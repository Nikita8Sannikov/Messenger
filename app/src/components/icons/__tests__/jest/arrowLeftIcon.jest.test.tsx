import { render, screen } from "@testing-library/react";
import ArrowLeftIconComponent from "@components/icons/arrowLeftIcon";

describe("ArrowLeftIcon", () => {
    test("renders correctly", () => {
        render(<ArrowLeftIconComponent />);
        const icon = screen.getByTestId("arrow-left-icon");
        expect(icon).toBeInTheDocument();
    });

    test("renders with custom size", () => {
        render(<ArrowLeftIconComponent size={32} />);
        const icon = screen.getByTestId("arrow-left-icon");
        expect(icon).toHaveAttribute("width", "32px");
        expect(icon).toHaveAttribute("height", "32px");
    });

    test("uses string size as is", () => {
        render(<ArrowLeftIconComponent size="2rem" />);
        const icon = screen.getByTestId("arrow-left-icon");
        expect(icon).toHaveAttribute("width", "2rem");
        expect(icon).toHaveAttribute("height", "2rem");
    });

    test("matches snapshot", () => {
        const { asFragment } = render(<ArrowLeftIconComponent></ArrowLeftIconComponent>);

        // Снимаем снапшот компонента
        expect(asFragment()).toMatchSnapshot();
    });
});