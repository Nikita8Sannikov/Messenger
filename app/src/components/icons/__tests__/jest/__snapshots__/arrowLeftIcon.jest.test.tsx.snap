// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowLeftIcon matches snapshot 1`] = `
<DocumentFragment>
  <svg
    data-testid="arrow-left-icon"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
      fill="currentColor"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="0"
    />
  </svg>
</DocumentFragment>
`;
